lib/three3d/core/BufferAttribute.dart

```diff
@@ -330,84 +330,97 @@ class BufferAttribute extends BaseBufferAttribute {
     // 	"array": this.array.sublist(0),
     // 	"normalized": this.normalized
     // };
-
-    var data = {
-      "itemSize": this.itemSize,
-      "type": this.array.runtimeType.toString(),
-      "array": this.array.sublist(0),
-      "normalized": this.normalized
+    Map<String, dynamic> data = {
+      "itemSize": itemSize,
+      "type": array.runtimeType.toString(), //.replaceAll('List', 'Array'),
+      "array": array.sublist(0),
+      "normalized": normalized
     };
 
-    if (this.name != null) data["name"] = this.name;
-    if (this.usage != StaticDrawUsage) data["usage"] = this.usage;
-    if (this.updateRange?["offset"] != 0 || this.updateRange?["count"] != -1)
-      data["updateRange"] = this.updateRange;
+    if (name != null) data["name"] = name;
+    if (usage != StaticDrawUsage) data["usage"] = usage;
+    if (updateRange?["offset"] != 0 || updateRange?["count"] != -1) {
+      data["updateRange"] = updateRange;
+    }
 
     return data;
   }
 }
 
-class Int8BufferAttribute extends BufferAttribute {
-  String type = "Int8BufferAttribute";
-
-  Int8BufferAttribute(array, itemSize, [bool normalized = false])
-      : super(array, itemSize, normalized) {}
+class Int8BufferAttribute extends BufferAttribute<Int8List> {
+  Int8BufferAttribute(Int8List array, int itemSize, [bool normalized = false])
```

Comment:
Hi @dmk-rib 
BufferAttribute
array should be Dart list like Int8List or native array from flutter_gl Int8Array

if use Int8List need memory copy every frame in some cases.

or we should use native array from flutter_gl

or we support two? let array is dynamic?

Reply:
do you have any good idea?

Reply:
In my opinion, it is worth using the native array then. I can prepare a version with NativeArraies and send you for review, is it ok?

Reply:
ok